### DISCUSSION TITLE
where to find the log file on android device

### DISCUSSION THREAD

**Discussion by Jzhenli** [beeware/toga#2909] [Q&A] [answered]
Hi, I run an simple toga apk with debug mode  on android device, after run some days, I want to check the log, where can find it?

**Comment by freakboy3742** [comment, marked as answer]
Android logs are obtained using the [Android Debug Bridge (adb)](https://developer.android.com/tools/adb); and, more specifically. the [logcat](https://developer.android.com/tools/logcat) tool that can be invoked over adb. ADB is installed as part of the Android SDK.

**Reply by mhsmith** [reply]
The [Android Studio Logcat window](https://developer.android.com/studio/debug/logcat) is also an option.

However, the log buffer size is limited, so don't expect to be able to read the log from several days ago. Even a few minutes may be too much in some cases.
